libparsec/crates/platform_http_proxy/src/windows.rs

```diff
@@ -0,0 +1,77 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use winreg::{enums::HKEY_CURRENT_USER, RegKey};
+
+use libparsec_types::anyhow;
+
+use crate::ProxyConfig;
+
+const PROXY_REGISTER_PATH: &str =
+    "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Internet Settings\\Connections";
+const PROXY_KEY: &str = "DefaultConnectionSettings";
+
+#[allow(dead_code)]
+const PROXY_AUTO_FLAG: u8 = 0b0001;
+/// Flag which determines if there is a proxy server
+const PROXY_SERVER_FLAG: u8 = 0b0010;
+/// Flag which determines if there is a proxy script
+const PROXY_SCRIPT_FLAG: u8 = 0b0100;
+#[allow(dead_code)]
+const PROXY_SKIP_LOCAL_FLAG: u8 = 0b1000;
+
+impl ProxyConfig {
+    fn with_proxy_script_url(mut self, url: String) -> Self {
+        self.proxy_script_url = url;
+        self
+    }
+
+    fn with_proxy_filter(mut self, filter: String) -> Self {
+        self.proxy_filter = filter.split(';').map(|x| x.to_string()).collect();
+        self
+    }
+
+    pub fn with_register(self) -> anyhow::Result<Self> {
```

Comment:
Should be called by `with_env_internal`

Reply:
tests using `with_env_internal` will fail on `windows` because we don't set `windows register` which need admin `privilege`